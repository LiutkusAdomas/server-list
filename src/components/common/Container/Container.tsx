import { ReactNode } from 'react';

type ContainerPropsType = {
  additional?: string;
  children?: ReactNode;
};

const Container: React.FC<ContainerPropsType> = ({ additional = '', children }) => {
  return (
    <section
      className={`bg-gray-50 py-12 h-fit px-4 sm:px-6 lg:px-8 rounded shadow-md ${additional}`}
    >
      <div className="w-full space-y-8">{children}</div>
    </section>
  );
};

export default Container;
