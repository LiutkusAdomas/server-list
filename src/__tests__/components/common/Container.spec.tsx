import { render } from '@testing-library/react';
import { describe, expect, it } from 'vitest';
import { Container } from '../../../components/common/Container/Container';

describe('Container component', () => {
  it('should render a container with text', () => {
    const { asFragment } = render(<Container>Test</Container>);
    expect(asFragment).toMatchSnapshot();
  });
  it('should render a container with text and additional classes', () => {
    const { asFragment } = render(<Container additional="w-full">Test</Container>);
    expect(asFragment).toMatchSnapshot();
  });
});
