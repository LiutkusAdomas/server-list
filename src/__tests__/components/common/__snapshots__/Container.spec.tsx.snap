// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Container component > should render a container with text 1`] = `[Function]`;

exports[`Container component > should render a container with text and additional classes 1`] = `[Function]`;
